fall-2020/navigation.py

```diff
@@ -87,9 +93,82 @@ def navigate(self):
         Returns: True if actuation requests were successful, False if not
         Note: It may be good to notify the user if something unexpected happens!
         """
-        pass
 
+        #Evaluate the differences of each position for x, y, and z
+
+        x = self.desired_position[0] - self.current_position[0]
+        y = self.desired_position[1] - self.current_position[1]
+        z = self.desired_position[2] = self.current_position[2]
+
+        """ The subsequent flow control follows the same format (until line 127). 
+
+        First it checks where the TBM is in relation to the target, such that if it is
+        behind it, to the left of it, to the right of it, above it, or below the target.
+
+        Based on the assessment for each difference for (x, y, z), it attempts to execute a movement,
+        then checks if the actuation was succesful! (Storing the bool value for x_act, y_act, z_act...)
+        
+        It is also important to note that the difference in x or y or z == 0, assume the actuation
+        was successful for that coordinate, as you would not need to steer in a direction!
+        """
+
+
+        #First check the x component and attempt actuation
+        if (x > 0):
+            x_act = self.steering.move_forward(x)
+
+        else:
+            x_act = self.steering.stop()
+        
+        #Next check the y component and attempt actuation
+        if (y > 0):
+            y_act = self.steering.move_right(y)
+
+        elif (y < 0):
+            y_act = self.steering.move_left(y)
+
+        else:
+            y_act = self.steering.stop()
 
+        #Lastly, check the z component and attempt actuation
+        if (z > 0):
+            z_act = self.steering.move_up(z)
+
+        elif (z < 0):
+            z_act = self.steering.move_down(z)
+
+        else:
+            z_act = self.steering.stop()
+
+        #Check if all actuations were successful! Return True if all were successful.
+        if x_act and y_act and z_act == True:
```

Comment:
The == True here is pretty unnecessary, and only gets "applied" to z_act anyways. 

Since they are all bools, ```x_act and y_act and z_act``` would do the trick. 

This is something I found strange in the past, say I wanted to compare 3 things to the string "paradigm". There is no syntactic sugar to do something like
```
if (a and b and c) == "Paradigm":
```
And instead you have to do
```
if a == "Paradigm and b == "Paradigm" and c == "Paradigm:"
```

Note:
There is a way to check if a bunch of things are either **all** true/false, or if **any** of the elements are true/false using the python functions any() and all().